### DISCUSSION TITLE
Configure dpr in srcset

### DISCUSSION THREAD

**Discussion by tkatchenko** [coreyward/gatsby-plugin-sanity-image#31] [Q&A] [answered]
Hello, I was wondering if there was a way to config `dpr` to 1 for every result in the `srcset`? I have some images that are coming in large on retina screens and while they look amazing they are quite large.

I've tried overriding with the `config` option but it looks like it doesn't affect the `srcset`.

Thanks!

**Comment by coreyward** [comment, marked as answer]
Are you specifying the `width` prop? The srcSet is built using that as a starting point, providing a range of multiples from it. The library uses DPR rather than modifying the width or height attributes passed to the Sanity Images API, so modifying the DPR to be 1 in all cases would be the same as not using a srcSet at all.

**Reply by tkatchenko** [reply]
Ah I see, as an example:

`<SanityImage {...image} width={1980} />`

Outputs:

`<img src="https://cdn.sanity.io/images/o5vscjl2/production/97bc9df4b4b3a6589a68a2d15a33179b0435dd7a-6000x4000.jpg?w=1980&amp;h=1320&amp;fm=webp" srcset="https://cdn.sanity.io/images/o5vscjl2/production/97bc9df4b4b3a6589a68a2d15a33179b0435dd7a-6000x4000.jpg?w=1980&amp;fm=webp&amp;dpr=0.5 990w,https://cdn.sanity.io/images/o5vscjl2/production/97bc9df4b4b3a6589a68a2d15a33179b0435dd7a-6000x4000.jpg?w=1980&amp;fm=webp&amp;dpr=0.75 1485w,https://cdn.sanity.io/images/o5vscjl2/production/97bc9df4b4b3a6589a68a2d15a33179b0435dd7a-6000x4000.jpg?w=1980&amp;fm=webp&amp;dpr=1 1980w,https://cdn.sanity.io/images/o5vscjl2/production/97bc9df4b4b3a6589a68a2d15a33179b0435dd7a-6000x4000.jpg?w=1980&amp;fm=webp&amp;dpr=1.5 2970w,https://cdn.sanity.io/images/o5vscjl2/production/97bc9df4b4b3a6589a68a2d15a33179b0435dd7a-6000x4000.jpg?w=1980&amp;fm=webp&amp;dpr=2 3960w" loading="lazy">`

Which on large monitors, not retina screens, displays a file that is [3960 × 2640 and ~750KB](https://cdn.sanity.io/images/o5vscjl2/production/97bc9df4b4b3a6589a68a2d15a33179b0435dd7a-6000x4000.jpg?w=1980&fm=webp&dpr=2). Looks like I've misunderstood how the DPR is used, but I guess I'd expect it to cap at the width of 1980?

Thanks for the reply and let me know if there's a work around, a way to set max width or otherwise. Library is great btw!

**Reply by tkatchenko** [reply]
To expand on this, it also appears that if I have a grid of smaller images they'll render out at twice the `width` attribute due to `srcset` looking at the window width.

**Reply by coreyward** [reply]
You should pass the actual display width of the image as your `width` prop. Say you show this image at 1000px wide. For someone on a high-res monitor with a device-pixel-ratio of 2.0, they would need a 2000px wide image to avoid each image pixel being represented by 4 physical pixels (i.e., to match the resolution of the surrounding content). For users with older, standard resolution displays, the browser should fetch the 1000px wide version of the image instead.

If you are encountering a situation where the browser is selecting a suboptimal option from the `srcSet`. you can use the `sizes` attribute to tell the browser how wide the image is expected to be displayed based on the viewport width. It's a good idea to always specify this in conjunction with `srcSet`, but it's important to make sure it's accurate or you can make matters worse.

**Reply by tkatchenko** [reply]
Thanks Corey! I'll look into `sizes` as it sounds like it will help do what I want, basically for photography rather than graphic imagery I'm happy outputting a DPR of 1 on screens with a high PPI density to reduce overall page size.
